flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/restart/RestartStrategyFactory.java

```diff
@@ -149,7 +149,7 @@ public static RestartStrategyFactory createRestartStrategyFactory(Configuration
 				}
 
 				// fallback in case of an error
-				return NoRestartStrategy.createFactory(configuration);
+				return NoOrFixedIfCheckpointingEnabledRestartStrategy.createFactory(configuration);
```

Comment:
Don't know why, but assumed the `default` branch is reached in case nothing was set in config. My mistake.

I've fixed it to differentiate the situation when `"none"` was set (this value is used across documentation, I think it should translate directly to `NoRestart`) and when the config was not set at all.